faskinner.lic

```diff
@@ -34,44 +38,64 @@ class FaSkinner
     elsif @priority == 'Both' || @priority == 'both'
       until DRSkill.getxp('First Aid') > 32  && DRSkill.getxp('Skinning') > 32
         train_skills
-      end 
+      end
       echo ("Exiting because First Aid AND Skinning locked!")
     else
       until DRSkill.getxp('First Aid') > 32  || DRSkill.getxp('Skinning') > 32
         train_skills
       end
       echo ("Exiting because First Aid OR Skinning locked!")
     end
+
+    do_exit
+
+  end
+
+  def do_exit
+    stow_item(@trainer, @trainer_container)
+    stow_item(@knife, @knife_container)
+    exit
   end
 
   def train_skills
-    case bput("skin my #{@trainer}",'You skillfully','A small blue-belly crocodile with prominently','You must be holding','The leather looks frayed','A small fuzzy caracal with tufted ears')
+    case bput("skin my #{@trainer}",'You skillfully','A small blue-belly crocodile with prominently','You must be holding','The leather looks frayed','A small fuzzy caracal with tufted ears','need to have a bladed instrument')
     when /You must be holding/
-      get_fred
+      get_item(@trainer, @trainer_container)
     when /The leather looks frayed/
-      echo ("No Charges!  Try again later.")
-      exit
-    end 
+      DRC.message("No Charges!  Try again later.")
+      do_exit
+    when /need to have a bladed instrument/
+      DRC.message("You need to either wear a skinning knife or hold a bladed weapon")
+      do_exit
+    end
 
     case bput("repair my #{@trainer}",'With some needle and thread','A small blue-belly crocodile with prominently','You must be holding','The leather looks frayed','A small fuzzy caracal with tufted ears')
     when /You must be holding/
-      get_fred
+      get_item(@trainer, @trainer_container)
     when /The leather looks frayed/
-      echo ("No Charges!  Try again later.")
-      exit
-    end 
-  end   
-
-  def get_fred
-    case bput("get my #{@trainer}",'You get','What were you referring')
-    when /What were you referring/
-      echo "OH MY DAMN - WHERE IS FRED!?"
+      DRC.message("No Charges!  Try again later.")
+      do_exit
+    end
+  end
+
+  def get_item(item, container = nil)
+    return unless item
+    return if DRCI.in_hands?(item)
+    unless DRCI.get_item(item, container)
+      DRC.message("Could not get '#{item}'!")
+      do_exit
     end
   end
+
+  def stow_item(item, container = nil)
+    return unless item
+    DRCI.put_away_item?(item, container)
+  end
+
```

Comment:
Note to self, https://github.com/rpherbig/dr-scripts/pull/4620#pullrequestreview-537768130 applies here, too